src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -230,12 +230,15 @@ Copyright (c) .NET Foundation. All rights reserved.
     <ItemGroup>
       <GenerateRuntimeConfigurationFilesInputs Include="$(_GenerateRuntimeConfigurationPropertyInputsCache)" />
 
-      <_GenerateRuntimeConfigurationPropertyInputsCacheToHash Include="$(TargetFramework)"/>
-      <_GenerateRuntimeConfigurationPropertyInputsCacheToHash Include="$(RuntimeIdentifier)"/>
+      <_GenerateRuntimeConfigurationPropertyInputsCacheToHash Include="@(AdditionalProbingPath)"/>
+      <_GenerateRuntimeConfigurationPropertyInputsCacheToHash Include="$(EnableDynamicLoading)"/>
       <_GenerateRuntimeConfigurationPropertyInputsCacheToHash Include="$(RollForward)"/>
-      <_GenerateRuntimeConfigurationPropertyInputsCacheToHash Include="$(AdditionalProbingPath)"/>
+      <_GenerateRuntimeConfigurationPropertyInputsCacheToHash Include="@(RuntimeHostConfigurationOption)"/>
```

Comment:
I think this should include the keys and values in the hash. Otherwise you're hashing just the keys, I think so the values could change and not change the hash.

See this for an example of how to achieve that: https://github.com/dotnet/sdk/blob/e605483699158a9202beb4eb9b4e18717c83a6f9/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.GenerateAssemblyInfo.targets#L126

Basically any metadata that is read by the task needs to be hashed.

We read at least both ItemSpec (Identity) and Value:

https://github.com/dotnet/sdk/blob/a5f5bb5183517cb301050288b25c9c727e52e8a6/src/Tasks/Microsoft.NET.Build.Tasks/GenerateRuntimeConfigurationFiles.cs#L283

https://github.com/dotnet/sdk/blob/a5f5bb5183517cb301050288b25c9c727e52e8a6/src/Tasks/Microsoft.NET.Build.Tasks/GenerateRuntimeConfigurationFiles.cs#L303

I recommend a test case that stresses this if I've correctly identified an issue here, because it's very subtle.

Reply:
Thanks for catching that, I added a test to cover that issue and updated the hash.